Fix get committees size using the state

StefanBratanov opened the pull request:
<!-- Thanks for sending a pull request! Please check out our contribution guidelines: -->
<!-- https://github.com/Consensys/teku/blob/master/CONTRIBUTING.md -->

## PR Description
Retrieve current epoch from `RecentChainData`. Otherwise, the second part of the code would never be reached.

## Fixed Issue(s)
fixes #9174 

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

StefanBratanov committed:
Fix getCommitteesSizeUsingTheState

lucassaldanha reviewed (approved):
Nice catch! There are some tests that might need an update.

rolfyone committed:
fix meaning of optional current epoch

also changed messages to debug as they're useless to users, which I had left previously but this will make more sense in debug.

The test cases had to be updated to return an epoch to make the code exercised.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

lucassaldanha reviewed (approved):
LGTM (for real this time) :)
